### PR TITLE
EPHEH-318: Pull labels from info file

### PR DESCRIPTION
**Pull request by drishu**

### CODE REVIEW

File: oe_color_scheme.module [hunk 1]
```diff
@@ -20,6 +20,12 @@ function oe_color_scheme_entity_view(array &$build, EntityInterface $entity) {
       continue;
     }
 
+    $name = $entity->get($field_name)->name;
+
+    if (empty($name)) {
+      continue;
+    }
+
     $build['#oe_color_scheme'] = $entity->get($field_name)->name;
```
Comment (brummbar) [inline oe_color_scheme.module]: Should we just use $name here?

File: src/Plugin/Field/FieldWidget/ColorSchemeWidget.php [hunk 2]
```diff
@@ -103,21 +104,17 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
...
-
-        return ucfirst($name);
-      }, $keys);
+    if (!isset($theme_info['color_scheme'])) {
+      return $options;
+    }
 
-      return array_combine($keys, $values);
+    foreach ($theme_info['color_scheme'] as $key => $label) {
+      $options[Html::getClass($key)] = strip_tags($label);
```
Comment (brummbar) [inline src/Plugin/Field/FieldWidget/ColorSchemeWidget.php]: Is this actually needed? In the end, we store the value in a string, and there's no need to transform it for the options field. Maybe we should enforce validation when fetching the options, even adding a hook_requirements that checks the current theme for the values.

File: tests/themes/oe_color_scheme_test_theme/oe_color_scheme_test_theme.info.yml [hunk 3]
```diff
@@ -4,5 +4,6 @@ base theme: false
 core_version_requirement: ^10
 
 color_scheme:
-  - pixy_dust
-  - powder_puff
+  pixy_dust: "Pixie Dust"
```
Comment (brummbar) [inline tests/themes/oe_color_scheme_test_theme/oe_color_scheme_test_theme.info.yml]: Quotes are not actually needed as per core.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: oe_color_scheme.module (hunk 1 above)
Comment (drishu) [reply oe_color_scheme.module]: y

File: src/Plugin/Field/FieldWidget/ColorSchemeWidget.php (hunk 2 above)
Comment (drishu) [reply src/Plugin/Field/FieldWidget/ColorSchemeWidget.php]: Any validation should be done in the requirements hook then.
I'm not convinced how validation is better in this case than processing, seems rather that we pass the cat from one to another.

Comment (brummbar) [reply src/Plugin/Field/FieldWidget/ColorSchemeWidget.php]: My main point is: do we need to have values in a specific format, eg no spaces, [, ], / (which is what Html::getClass does)?
If we don't, I would rather expect the values to be the ones declared, and not tamper with them.
If we do, then I would set validation on them, and exclude the ones that don't respect the format, but still avoid changing the values.

Comment (drishu) [reply src/Plugin/Field/FieldWidget/ColorSchemeWidget.php]: I would argue that yes, we need getClass type values for the color schemes, it needs to be opinionated that way, as a color scheme is a front thing, not just a abstract concept, so its output needs to be front usable out of the bat.

Comment (drishu) [reply src/Plugin/Field/FieldWidget/ColorSchemeWidget.php]: I've implemented the filtering of options

**Pull request closed without merging**